snapcraft/storeapi/_snap_index_client.py

```diff
@@ -46,40 +63,37 @@ def get_default_headers(self):
 
         return headers
 
-    def get_package(self, snap_name, channel=None, arch=None):
-        """Get the details for the specified snap.
+    def get_info(self, snap_name: str, *, arch: str = None) -> SnapInfo:
+        """Get the information for the specified snap.
 
         :param str snap_name: Name of the snap.
-        :param str channel: Channel of the snap.
         :param str arch: Architecture of the snap (none by default).
 
-        :return Details for the snap.
-        :rtype: dict
+        :return information for the snap.
+        :rtype: SnapInfo
         """
         headers = self.get_default_headers()
         headers.update(
             {
                 "Accept": "application/hal+json",
```

Comment:
v2 responses are technically only `application/json`, we can drop the `hal+` part